logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+			rc = sqlite3_exec(p_file_info->db, "END TRANSACTION;", NULL, NULL, NULL);
+			if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+			
+			/* Replace the maximum number with 0 in BLOB files. */
+			sprintf(old_path, "%s" BLOB_STORE_FILENAME ".%d", p_file_info->db_dir, BLOB_MAX_FILES);
+			sprintf(new_path, "%s" BLOB_STORE_FILENAME ".%d", p_file_info->db_dir, 0);
+			rc = uv_fs_rename(writer_loop, &rename_req, old_path, new_path, NULL);
+			if (unlikely(rc)) fatal_libuv_err(rc, __LINE__);
+			uv_fs_req_cleanup(&rename_req);
+			
+			/* (a) Update blob_write_handle_offset, (b) truncate new write-to BLOB, 
+			 * (c) update filesize of truncated BLOB in SQLite DB, (d) delete
+			 * respective logs in LOGS_TABLE for the truncated BLOB and (e)
+			 * reset blob_filesize */
+			/* (a) */ 
+			p_file_info->blob_write_handle_offset = p_file_info->blob_write_handle_offset == 1 ? BLOB_MAX_FILES : p_file_info->blob_write_handle_offset - 1;
+			/* (b) */ 
+			uv_fs_t trunc_req;
+			rc = uv_fs_ftruncate(writer_loop, &trunc_req, p_file_info->blob_handles[p_file_info->blob_write_handle_offset], 0, NULL);						
+			if (unlikely(rc)) fatal_libuv_err(rc, __LINE__);
+			uv_fs_req_cleanup(&trunc_req);
+			/* (c) */ 
+			rc = sqlite3_bind_int(stmt_blobs_set_zero_filesize, 1, p_file_info->blob_write_handle_offset);
+            if (rc != SQLITE_OK) fatal_sqlite3_err(rc, __LINE__);
+            rc = sqlite3_step(stmt_blobs_set_zero_filesize);
+			if (unlikely(rc != SQLITE_DONE)) fatal_sqlite3_err(rc, __LINE__);
+			rc = sqlite3_reset(stmt_blobs_set_zero_filesize);
+			if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+			/* (d) */
+			rc = sqlite3_bind_int(stmt_logs_delete, 1, p_file_info->blob_write_handle_offset);
+            if (rc != SQLITE_OK) fatal_sqlite3_err(rc, __LINE__);
+            rc = sqlite3_step(stmt_logs_delete);
+			if (unlikely(rc != SQLITE_DONE)) fatal_sqlite3_err(rc, __LINE__);
+			rc = sqlite3_reset(stmt_logs_delete);
+			if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+			/* (e) */
+			blob_filesize = 0;
+
+		}
+
+		/* Update total disk usage of all BLOBs for this log source */
+		rc = sqlite3_step(stmt_blobs_get_total_filesize);
+		if (unlikely(rc != SQLITE_ROW)) fatal_sqlite3_err(rc, __LINE__);
+		p_file_info->blob_total_size = sqlite3_column_int64(stmt_blobs_get_total_filesize, 0);
+		// debug(D_LOGS_MANAG, "p_file_info->blob_total_size: %lld for: %s\n", p_file_info->blob_total_size, p_file_info->filename);
+		rc = sqlite3_reset(stmt_blobs_get_total_filesize);
+		if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+
+		// TODO: Can db_release_lock(p_file_info->db_mut) be moved before if(blob_filesize > p_file_info-> blob_max_size) ?
+		db_release_lock(p_file_info->db_mut);
+		uv_sleep(p_file_info->buff_flush_to_db_interval * MS_IN_SEC);
+	}
+}
+
+/**
+ * @brief Process the events of the uv_loop_t related to the DB API
+ */
```

Comment:
Please, convert next two functions for `inline`.

Reply:
Done.